library/acl.inc

```diff
@@ -828,7 +828,11 @@
             return false;
         elseif($count_user_allowed > 0)
             return true;
-        elseif($count_group_denied > 0)
+        /*
+            We want to include user that belongs to group A and group B
+            and return true in case of group A have permissions and group B does not have permissions
+        */
+        elseif($count_group_denied > 0 && $count_group_allowed < 1)
```

Comment:
Yes you are right that the function can be written more efficiently and should be re-examined ,
but I think meanwhile we should apply this , until the code is re factored.